v2/internal/reconcilers/reconcile_policy.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package reconcilers
+
+import "github.com/pkg/errors"
+
+// ReconcilePolicyAnnotation describes the reconcile policy for the resource in question.
+// A reconcile policy describes what action (if any) the operator is allowed to take when
+// reconciling the resource.
+// If no reconcile policy is specified, the default is "run"
+const ReconcilePolicyAnnotation = "serviceoperator.azure.com/reconcile-policy"
```

Comment:
I'm not _totally_ sold on this name. Open to better ones if people have them. Other ideas I had:
`azure-interaction-policy`, `reconcile-mode`?

Similarly I am not totally sold on the values. Would `reconcile-policy: act|observe|skip-delete` be better?

Reply:
Or `abandon` as @jonnylangefeld suggests below for the `skip-delete` case?

Reply:
Ok I like: `reconcile-policy`: `manage|skip|detach-on-delete` I think. Any thoughts?

Reply:
How about ? mimicking Azure action [types](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-definitions#actions-format)
```
"serviceoperator.azure.com/object-access-type": "*/read/write/delete"
```

Reply:
I'm not sure we need this level of flexibility - what would it mean for ASO to have `read|delete` as the policy?

Reply:
Agree @theunrepentantgeek - this is part of the reason why it feels better IMO to have a single `policy` rather than a `modification-policy` and a `delete-policy`, because it'd be weird to allow deletes but not writes. The 3 values that seem to make sense are:
`read|write|delete` == my proposed `manage`
`read|write` == my proposed `detach-on-delete`
`read` == my proposed `skip`.